packages/trace-viewer/src/sw/main.ts

```diff
@@ -43,9 +43,10 @@ async function loadTrace(traceUrl: string, traceFileName: string | null, client:
   const clientId = client?.id ?? '';
   let data = clientIdToTraceUrls.get(clientId);
   if (!data) {
-    let traceViewerServerBaseUrl = new URL('../', client?.url ?? self.registration.scope);
-    if (traceViewerServerBaseUrl.searchParams.has('server'))
-      traceViewerServerBaseUrl = new URL(traceViewerServerBaseUrl.searchParams.get('server')!, traceViewerServerBaseUrl);
+    const clientURL = new URL(client?.url ?? self.registration.scope);
+    let traceViewerServerBaseUrl = new URL('../', clientURL);
```

Comment:
IIUC, we are trying to support relative `server` url here? If so, perhaps it should be relative to the `clientURL` instead of `'../'`? I find this contract to be harder to understand.

Basically:

```js
const traceViewerServerUrl = new URL(clientURL.searchParams.get('server') ?? '../', clientURL);
```

Reply:
That works, sure! We're not currently using relative `server` anyways, happy to change this to anything.

Reply:
done, implemented your recommendation